src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -183,66 +176,58 @@ function addScriptTagsToDocument(resourceLoader: WebAssemblyResourceLoader) {
     .keys(resourceLoader.bootConfig.resources.runtime)
     .filter(n => n.startsWith('dotnet.') && n.endsWith('.js'))[0];
   const dotnetJsContentHash = resourceLoader.bootConfig.resources.runtime[dotnetJsResourceName];
-  const scriptElem = document.createElement('script');
-  scriptElem.src = `_framework/${dotnetJsResourceName}`;
-  scriptElem.defer = true;
-
-  // For consistency with WebAssemblyResourceLoader, we only enforce SRI if caching is allowed
-  if (resourceLoader.bootConfig.cacheBootResources) {
-    scriptElem.integrity = dotnetJsContentHash;
```

Comment:
There is no way how to check SRI via dynamic `import` AFAIK.
There are already other dynamic imports in the codebase.

Reply:
I think we might want to produce a single bundle that incorporates dotnet.js into blazor.webassembly.js we don't want to loose the SRI bit (even though it's not that important for scripts hosted locally) and consuming the module and producing a single bundle is more natural. (That's the direction that we were headed).

That said, I think the current changes are a good first step, and we need to think a few things through to achieve the rest:
* Single bundle (we need to think about how this plays with relinking of dotnet.js)
* Global state: I think we need to figure out the interface between blazor/mono when we use modules. Removing any global state is something we want to achieve, that said, I think we need to make it configurable for backwards compatibility. 
  * I think this applies to both Blazor and mono globals
* Contract between Blazor/mono: I believe we were doing some things through global callbacks, it would be neat if we are able to get rid of those.

Reply:
- runtime doesn't have any global state anymore, it's fully modularized. I should probably write unit test which creates multiple runtimes on same page to prove it.
- if you set `disableDotnet6Compatibility: true` you will see if any of your tests fail, because blazor still calls runtime via global namespace.

Reply:
Note that packing dotnet into `blazor.webassembly.js` is possible, but it has 2 consequences
- it tries to find `dotnet.wasm` in same directory as the bundle, which is not the same as location of eventual npm package. In case of Blazor this is probably non-issue as you have your cache anyway.
- AOT/native linking scenarios related to `dotnet.wasm` and `dotnet.js` need to be designed.

Anyway, trivial [webpack sample here](https://github.com/dotnet/runtime/tree/3591addba0a7394123fd522c9951d52c5854f261/src/mono/sample/wasm/browser-webpack)

Reply:
Let's create separate issue and merge this one. I would like to proceed with ES6 switch and let you do Blazor cleanup.

Reply:
@SteveSandersonMS do you think that the `modulepreload` would prevent `import` from loading file with different hash ?

Reply:
The sample at https://stackoverflow.com/a/63029759 appears to show that it does prevent loading a file with a wrong hash.

Reply:
I tested it and it works. Thanks! 
Code is updated and I do it only for chromium.